jsbits/jsstring.js

```diff
@@ -853,37 +835,39 @@ function h$jsstringDecW64(hi,lo) {
     return '' + x2 + h$jsstringDecIPadded6(x1);
 }
 
-#if __GLASGOW_HASKELL__ >= 800
-function h$jsstringHexInteger(val) {
-  TRACE_JSSTRING("hexInteger");
-  if(IS_INTEGER_S(val)) {
-    return '' + INTEGER_S_DATA(val);
-  } else {
-    // we assume it's nonnegative. this condition is checked by the Haskell code
-    return h$ghcjsbn_showBase(INTEGER_J_DATA(val), 16);
+function h$jsstringHexBigNat(positive,x) {
+  TRACE_JSSTRING("hexBigNat");
+  var v = "";
+  var i = x.u1.length - 1;
+  while (i >= 0) {
+    if (x.u1[i] !== 0) {
+      break;
+    } else {
+      i--;
+    }
   }
-}
-#else
-function h$jsstringHexInteger(val) {
-  TRACE_JSSTRING("hexInteger");
-  if(IS_INTEGER_S(val)) {
-    return '' + INTEGER_S_DATA(val);
-  } else {
-    return INTEGER_J_DATA(val).toRadix(16);
+  if (i >= 0) {
+    v += x.u1[i].toString(16);
+    i--;
+  }
+  for (; i >= 0; i--) {
+    v += x.u1[i].toString(16).padStart(4, '0');
   }
+  return positive ? v : "-"+v;
 }
-#endif
 
 function h$jsstringHexI64(hi,lo) {
-    var lo0 = lo<0 ? lo+4294967296 : lo;
-    if(hi === 0) return lo0.toString(16);
-    return ((hi<0)?hi+4294967296:hi).toString(16) + h$jsstringHexIPadded8(lo0);
+    TRACE_JSSTRING("hexI64: " + hi + " " + lo);
+    var sign = (hi >>> 31) ? '-' : '';
+    var lo0 = sign ? ~(lo - 1) >>> 0 : lo;
```

Comment:
please add a comment about `>>> 0`. Something like:

```suggestion
    // unsigned right bitshift by zero to convert to UInt32
    var lo0 = sign ? ~(lo - 1) >>> 0 : lo;
```